openedx/core/djangoapps/xblock/runtime/runtime.py

```diff
@@ -101,6 +102,7 @@ def __init__(self, system: XBlockRuntimeSystem, user: UserType | None):
             mixins=(
                 LmsBlockMixin,  # Adds Non-deprecated LMS/Studio functionality
                 XBlockShim,  # Adds deprecated LMS/Studio functionality / backwards compatibility
+                InheritanceMixin,
```

Comment:
@Cup0fCoffee, I need some help understanding the implications of this change.

@bradenmacdonald, @ormsbee, @kdmccormick, do you know why the `InheritanceMixin` was not explicitly listed in the mixins for the new runtime?

Reply:
> do you know why the InheritanceMixin was not explicitly listed in the mixins for the new runtime?

The new runtime is only used for content libraries and LabXchange, both of which have very flat content trees, sp I think that we didn't have any need for it.

Reply:
Hey @bradenmacdonald @Cup0fCoffee @Agrendalath! Just checking in on this.

Reply:
There is a long-term interest in flattening out XBlocks. All XBlocks would be leaf nodes, and structure/dynamicism would be handled by separate subsystem(s) in defined in openedx-learning. Leaving InheritanceMixin out of the new runtime was a nice nod towards this. This [ADR describes the rationale](https://github.com/openedx/edx-platform/blob/master/docs/decisions/0006-role-of-xblock.rst), although it only goes as far as saying that the Unit should be the highest-level xblock, rather than getting rid of inheritance entirely.

I know there are still several blocks today that rely on parent-child relationships: chapter/sequence (obviously), library_content, split_test, conditional, problem_builder. So, it might be inevitable that we add InhertianceMixin the new runtime. @Cup0fCoffee can you help me understand why this PR in particular needs to add the mixin?

Reply:
@kdmccormick The reasoning behind it is that @Agrendalath noticed that `ProblemBlock` implements several methods that are also implemented in the `InheritanceMixin` exactly the same way. Since there was an assumption that XBlocks inherit from the `InheritanceMixin`, the methods that already implemented by the `InheritanceMixin` can be removed from the `ProblemBlock`. So in this PR I attempted to remove the `close_date` and `is_past_due` methods.

Once I removed the methods, some tests started failing with errors like `AttributeError: 'ProblemBlock' object has no attribute 'is_past_due'`, because runtimes in those tests were not adding `InheritanceMixin` to the list of mixins, so the XBlocks that were being created were not inheriting from it. So for each failing test, I started tracking down the runtimes that were being used, and adding the mixin to the list until all the tests with this specific error started passing.

You can find more details in the following thread - https://github.com/openedx/edx-platform/pull/34320#discussion_r1519968408.

Reply:
Thanks for you explanation @Cup0fCoffee , and sorry it took so long for me to get back to you.

I was confused for a while because I had forgotten that InheritanceMixin doesn't actually add inheritance capabilities... rather, it just _defines fields which the modulestore runtime treats as inheritable_.

Given that, I am OK with adding the mixin to Learning Core runtime, but with the knowledge that no field inhertiance will actually happen in that runtime. Could you add a note to InheritanceMixin explaining this? Even better, if you have time, I think it would be good to rename the mixin to InheritableFieldsMixin.

That's my sense at least. @bradenmacdonald , does that sound right to you too?

Reply:
That sounds reasonable @kdmccormick.